### DISCUSSION TITLE
05_drivers_gpio_uart use a USB storage device intead of MicroSD card for rpi3?

### DISCUSSION THREAD

**Discussion by ChocolateLoverRaj** [rust-embedded/rust-raspberrypi-OS-tutorials#221] [Tutorial Questions] [answered]
I don't have an empty MicroSD card but I have an empty USB drive. Can I use that instead of a MicroSD card? I have a Raspberry Pi 3b.

**Comment by BartMassey** [comment, marked as answer]
Haven't looked at this guide in a while, but I think you may run into issues later on. MicroSD card is I2C which is easy to talk to, USB is quite a bit more of an adventure…

**Reply by ChocolateLoverRaj** [reply]
Maybe booting from USB can be a more advanced part of the tutorial later on (I am only on 07 rn so I don't even know if the tutorial gets to USB later)
